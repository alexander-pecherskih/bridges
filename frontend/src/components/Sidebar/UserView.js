import React from 'react'
import {Link} from 'react-router-dom'

const UserView = ({ userInfo }) => {
    return <div className="user-view">
        <div className="background"></div>
        <Link to="/profile"><img className="circle user-view__avatar" src={ userInfo.avatar }  alt={ userInfo.name }/></Link>
        <Link to="/profile"><span className="white-text user-view__name">{ userInfo.name }</span></Link>
    </div>
}

export default UserView