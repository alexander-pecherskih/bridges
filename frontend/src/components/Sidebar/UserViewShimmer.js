import React from 'react'
import {Link} from 'react-router-dom'

const imageStyle = {
    width: '64px',
    height: '64px',
    backgroundColor: 'gray',
}
const textStyle = {
    height: '1rem',
    backgroundColor: 'gray',
    margin: '22px 0 11px 0',
    borderRadius: '5px',
}

const UserViewShimmer = () => {
    return <div className="user-view">
        <div className="background" />
        <div className="circle" style={ imageStyle } />
        <div className="shimmer-text" style={ textStyle } />
    </div>
}

export default UserViewShimmer