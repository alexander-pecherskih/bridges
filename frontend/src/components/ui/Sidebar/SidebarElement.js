import React from 'react'

import styles from './styles/SidebarElement.module.css'

const SidebarElement = ({ children }) => (
  <div className={styles.sidebarElement}>{children}</div>
)

export default SidebarElement
