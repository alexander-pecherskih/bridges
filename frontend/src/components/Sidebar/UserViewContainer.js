import React, {useEffect} from 'react'
import { compose } from 'redux'
import { connect } from 'react-redux'

import UserInfoService from '../../services/UserInfoService'
import { getUserInfo } from '../../store/actions/userInfo'

import UserView from './UserView'
import UserViewShimmer from './UserViewShimmer'

const UserViewContainer = ({ user, loading, error, getUserInfo }) => {
    // const userInfo = UserInfoService.defaultUserInfo
    useEffect(getUserInfo, [])

    if (!user){
        return <UserViewShimmer />
    }

    return <UserView userInfo={ user } />
}

const mapStateToProps = ({ userInfo: { user, loading, error } }) => {
    return { user, loading, error }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getUserInfo: getUserInfo(dispatch),
    }
}

export default compose(
    connect(mapStateToProps, mapDispatchToProps)
)(UserViewContainer)