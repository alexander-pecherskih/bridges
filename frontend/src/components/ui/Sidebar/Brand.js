import React from 'react'
import PropTypes from 'prop-types'

import SidebarElement from './SidebarElement'
import Logo from '../Logo/Logo'

import styles from './styles/Brand.module.css'

const Brand = ({ caption }) => {
  return (
    <SidebarElement>
      <div className={styles.brand}>
        <Logo size={32} />
        <div className={styles.caption}>{caption}</div>
      </div>
    </SidebarElement>
  )
}
Brand.propTypes = {
  caption: PropTypes.string.isRequired,
}

export default Brand
