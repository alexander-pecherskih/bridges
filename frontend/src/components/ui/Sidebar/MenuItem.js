import React from 'react'
import SidebarElement from './SidebarElement'

import styles from './styles/MenuItem.module.css'
import { NavLink } from 'react-router-dom'

const MenuItem = ({ caption, url }) => {
  return (
    <SidebarElement>
      <NavLink className={styles.menuItem} to={url}>{caption}</NavLink>
    </SidebarElement>
  )
}

export default MenuItem
