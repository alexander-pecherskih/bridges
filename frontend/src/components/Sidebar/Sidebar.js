import React, { useEffect, useRef } from 'react'
import { Link } from 'react-router-dom'
import M from 'materialize-css'

import UserViewContainer from './UserViewContainer'
// import Icon from '../Icon'

const Divider = () => <li><div className="divider"></div></li>

const Sidebar = () => {
  const onMount = () => { M.Sidenav.init(sidebar) }
  const sidebar = useRef(null)

  useEffect(onMount, [])

  return <>
    <ul className="sidenav sidenav-fixed sidebar" ref={ sidebar } >
      <li>
        <UserViewContainer />
      </li>
      <li><Link to="/">
        {/* <Icon icon="home" /> */}
                Задачи
      </Link></li>
      <Divider/>
      <li><Link to="/processes">Процессы</Link></li>
      <li><Link to="/settings">Настройки</Link></li>
      {/* eslint-disable-next-line */}
      {/* <li><a href="#">Third Link With Waves</a></li> */}
    </ul>
    {/* eslint-disable-next-line */}
    {/* <a href="#" data-target="slide-out" className="sidenav-trigger"><i className="material-icons">menu</i></a> */}
  </>
}

export default Sidebar
