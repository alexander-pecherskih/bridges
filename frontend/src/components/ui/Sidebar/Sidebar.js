import React from 'react'

import styles from './styles/Sidebar.module.css'
import MenuItem from './MenuItem'
import Title from './Title'
import Brand from './Brand'

const Sidebar = ({ visible }) => {
  return (
    <>
      <Brand caption="Bridges" />
      <nav className={`${styles.sidebar} ${visible ? '' : styles.hidden}`}>
        <Title caption="processes" />
        <MenuItem caption="Tickets" url="/" />
        <Title caption="config" />
        <MenuItem caption="Process List" url="/process-list" />
      </nav>
    </>
  )
}

export default Sidebar
