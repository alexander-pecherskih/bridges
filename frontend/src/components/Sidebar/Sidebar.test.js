import React from 'react'
import { shallow } from 'enzyme'

import Sidebar from './Sidebar'

const defaultProps = {
    identity: {
        user: 'John',
    }
}

describe('<Sidebar />', () => {
    const wrapper = shallow(<Sidebar { ...defaultProps } />)

    it('Sidebar Have a User Name', () => {
        expect(wrapper.find('.user-view .name').text()).toBe(defaultProps.identity.user)
    })
})