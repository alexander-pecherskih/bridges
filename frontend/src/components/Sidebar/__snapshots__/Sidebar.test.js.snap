// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders properly 1`] = `ShallowWrapper {}`;
